source/MRMesh/MRObjectMesh.cpp

```diff
@@ -200,6 +205,30 @@ std::shared_ptr<ObjectMesh> merge( const std::vector<std::shared_ptr<ObjectMesh>
                 hasVertColorMap = true;
             if ( !obj->getFacesColorMap().empty() )
                 hasFaceColorMap = true;
+
+            const auto& textures = obj->getTextures();
+            if ( needSaveTexture )
+            {
+                if ( textures.empty() )
+                {
+                    needSaveTexture = false;
+                    continue;
+                }
+
+                numTexture += textures.size();
+                numObject++;
+
+                for ( const auto& texture : textures )
+                {
+                    if ( resolution.x == -1 )
+                        resolution = texture.resolution;
+                    else if ( resolution != texture.resolution )
+                    {
+                        needSaveTexture = false;
+                        break;
+                    }
+                }
```

Comment:
checking first one should be enough, as far as all textures in one object should be same size